File: python/2011/rise-of-the-planet-of-the-apes-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Rise of the Planet of the Apes to the database
movies.insert(
    title="Rise of the Planet of the Apes", 
    year=2011, 
    plot="A substance, designed to help the brain repair itself, gives rise to a super-intelligent chimp who leads an ape uprising.", 
    rating=7.6
)

# Confirm that the movie was added
movie = movies.select(
    title="Rise of the Planet of the Apes", 
    year=2011
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    

File: csharp/2011/rise-of-the-planet-of-the-apes-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Rise of the Planet of the Apes" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Rise of the Planet of the Apes",
                year: 2011,
                plot: "A substance, designed to help the brain repair itself, gives rise to a super-intelligent chimp who leads an ape uprising.",
                rating: 7.6
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Rise of the Planet of the Apes",
                year: 2011
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}
